eventsources/sources/github/start.go

```diff
@@ -124,182 +125,183 @@ func (router *Router) HandleRoute(writer http.ResponseWriter, request *http.Requ
-	hook, _, err := router.githubClient.Repositories.CreateHook(ctx, githubEventSource.Owner, githubEventSource.Repository, router.hook)
-	if err != nil {
-		// Continue if error is because hook already exists
-		er, ok := err.(*gh.ErrorResponse)
-		if !ok || er.Response.StatusCode != http.StatusUnprocessableEntity {
-			return errors.Errorf("failed to create webhook. err: %+v", err)
+		logger.Info("configuring GitHub hook...")
+		formattedURL := common.FormattedURL(githubEventSource.Webhook.URL, githubEventSource.Webhook.Endpoint)
+		hookConfig := map[string]interface{}{
+			"url": &formattedURL,
 		}
-	}
-
-	// if hook alreay exists then CreateHook returns hook value as nil
-	if hook == nil {
-		logger.Info("GitHub hook for the repository already exists, trying to use the existing hook...")
-		ctx, cancel = context.WithTimeout(context.Background(), 20*time.Second)
-		defer cancel()
-
-		hooks, _, err := router.githubClient.Repositories.ListHooks(ctx, githubEventSource.Owner, githubEventSource.Repository, nil)
-		if err != nil {
-			return errors.Errorf("failed to list existing webhooks. err: %+v", err)
+		if githubEventSource.ContentType != "" {
+			hookConfig["content_type"] = githubEventSource.ContentType
 		}
-
-		hook = getHook(hooks, formattedURL, githubEventSource.Events)
-		if hook == nil {
-			return errors.New("failed to find existing webhook")
+		if githubEventSource.Insecure {
+			hookConfig["insecure_ssl"] = "1"
+		} else {
+			hookConfig["insecure_ssl"] = "0"
+		}
+		if router.hookSecret != "" {
+			hookConfig["secret"] = router.hookSecret
 		}
-	}
-
-	if githubEventSource.WebhookSecret != nil {
-		// As secret in hook config is masked with asterisk (*), replace it with unmasked secret.
-		hook.Config["secret"] = hookConfig["secret"]
-	}
-
-	router.hook = hook
-	logger.Info("GitHub hook has been successfully set for the repository")
-
-	return nil
-}
-
-// PostInactivate performs operations after the route is inactivated
-func (router *Router) PostInactivate() error {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
-
-	githubEventSource := router.githubEventSource
-
-	if githubEventSource.APIToken == nil || githubEventSource.Webhook.URL == "" {
-		logger := router.route.Logger.With(
-			"repository", githubEventSource.Repository,
-		)
 
-		logger.Info("no api credential or webhook url specified, skipping webhook deletion...")
-		return nil
-	}
+		logger.Info("setting up client for GitHub...")
+		client := gh.NewClient(PATTransport.Client())
 
-	if githubEventSource.DeleteHookOnFinish {
-		logger := router.route.Logger.With(
-			"repository", githubEventSource.Repository,
-			"hook-id", *router.hook.ID,
-		)
+		logger.Info("setting up base url for GitHub client...")
+		if githubEventSource.GithubBaseURL != "" {
+			baseURL, err := url.Parse(githubEventSource.GithubBaseURL)
+			if err != nil {
+				return errors.Errorf("failed to parse github base url. err: %s", err)
```

Comment:
suggestion - I prefer `fmt.Errorf` its both core Go, and less verbose than `errors.Errorf`

Reply:
done.